.circleci/config.yml

```diff
@@ -32,6 +33,43 @@ jobs:
       - run: xcodebuild -version
       - run: xcodebuild test -scheme HoneycombTests -sdk iphonesimulator17.5 -destination 'OS=17.5,name=iPhone 15'
 
+  smoke_tests:
+    macos:
+      xcode: 15.4.0
+    resource_class: macos.m1.medium.gen1
+
+    steps:
+      - attach_workspace:
+          at: ./
+      - macos/preboot-simulator:
+          version: "17.5"
+          platform: "iOS"
+          device: "iPhone 15"
+      - checkout
+      - bats/install
+      - run:
+          name: What's the BATS?
+          command: |
+            which bats
+            bats --version
+      - run: xcodebuild -version
+      - run:
+          name: Install OTel Collector
+          environment:
+            COLLECTOR_VERSION: "0.111.0"
+          command: |
+            curl --proto '=https' --tlsv1.2 -fOL https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v${COLLECTOR_VERSION}/otelcol_${COLLECTOR_VERSION}_darwin_arm64.tar.gz
```

Comment:
I was going to suggest using docker or caching the download could be good but the smoke tests take 10mins to run so probably not worth it 😂

Reply:
That was the original plan 😭 but I found it was really hard to get docker working on the CI mac machine and the collector setup itself is super fast. The thing that takes the longest is running the iOS app, which we can probably make faster with some CircleCI tricks.